### PR TITLE
Fix Backup cookie

### PR DESCRIPTION
**Pull request by dauglyon**
This PR fixes the backup cookie issue. It fixes the cookie configuration, adds tests specific to the backup cookie, and adds a missing condition that obfuscated the configuration error and accidentally worked only on login/logout or when a backup cookie was already present.

### CODE REVIEW

File: scripts/build_deploy.ts [hunk 1]
```diff
@@ -60,6 +61,7 @@ const setEnvironment = (
     REACT_APP_KBASE_DOMAIN: domain,
     REACT_APP_KBASE_LEGACY_DOMAIN: legacy,
     REACT_APP_KBASE_BACKUP_COOKIE_NAME: backupCookie?.name || '',
+    REACT_APP_KBASE_BACKUP_COOKIE_DOMAIN: backupCookie?.domain || '',
```
Comment (dauglyon) [inline scripts/build_deploy.ts:64]: the main issue

File: src/features/auth/authSlice.test.tsx
Comment (dauglyon) [inline src/features/auth/authSlice.test.tsx:1]: Added the ability to test cookies independently, added new backup cookie tests

File: src/features/auth/hooks.ts
Comment (dauglyon) [inline src/features/auth/hooks.ts:1]: This should catch a mismatched configuration in the future.

**Review by dauglyon — COMMENTED** [review]
left some comments explaining which change does what

**Review by briehl — APPROVED** [review]
Looks good to me. @dakotablair do you want to chime in?

**Review by dakotablair — APPROVED** [review]

**Pull request merged**